<launch>

    <!-- important generally, but specifically utilised by the current app manager -->
    <param name="robot/name" value="$(optenv ROBOT turtlebot)"/>
    <param name="robot/type" value="turtlebot"/>


    <param name="robot_description" command="cat $(find ros_gibson_environment)/robot_models/turtlebot.urdf"/>

    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
    </node>

    <!--
        semantic_visualization_mode parameter possible values:
            - distinctive: the environment assigns a random distinctive color to each semantic instance.
                            This mode renders intuitive colorful semantic map.
            - label_to_rgb: the environment assigns a semantic label to each object instance and then converts each label in an rgb color.
                            the rendered frame can be directly consumed as the semantically labelled pixel maps
                            but the frame itself usually looks dark to human eyes.
     -->
    <node name="ros_gibson_environment_simulator" pkg="ros_gibson_environment" type="turtlebot_gibson_simulator.py" output="screen">
        <param name="environment" value="house1" />
        <param name="semantic_visualization_mode" value="label_to_rgb" />
        <param name="resolution" value="256" />
    </node>

    <node name="gibson_clock" pkg="ros_gibson_environment" type="simulation_clock.py">
    </node>

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
        <param name="publish_frequency" type="double" value="5.0" />
    </node>

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find ros_gibson_environment)/params/rviz_gmapping.rviz" />

    <!-- Velocity muxer -->
    <node pkg="nodelet" type="nodelet" name="mobile_base_nodelet_manager" args="manager"/>
    <node pkg="nodelet" type="nodelet" name="cmd_vel_mux"
          args="load yocs_cmd_vel_mux/CmdVelMuxNodelet mobile_base_nodelet_manager">
        <param name="yaml_cfg_file" value="$(find ros_gibson_environment)/params/turtlebot/mux_params.yaml" />
        <remap from="cmd_vel_mux/output" to="mobile_base/commands/velocity"/>
    </node>
</launch>